src/d/actor/d_a_ks.cpp

```diff
@@ -5,125 +5,1613 @@
 
 #include "d/actor/d_a_ks.h"
 #include "d/d_procname.h"
+#include "d/d_com_inf_game.h"
+#include "d/d_s_play.h"
+#include "f_op/f_op_camera.h"
+#include "d/d_snap.h"
+#include "d/actor/d_a_player_main.h"
+#include "d/actor/d_a_gm.h"
+#include "d/res/res_ks.h"
+
+#include "weak_bss_936_to_1036.h" // IWYU pragma: keep
+#include "weak_data_1811.h" // IWYU pragma: keep
+
+int KS_ALL_COUNT;
+int KUTTUKU_ALL_COUNT;
+BOOL HEAVY_IN;
+int GORON_COUNT;
 
 /* 00000078-000002CC       .text draw_SUB__FP8ks_class */
-void draw_SUB(ks_class*) {
-    /* Nonmatching */
+void draw_SUB(ks_class* i_this) {
+    J3DModel* pBodyModel = i_this->mpBodyMorf->getModel();
+    J3DModel* pEyeModel = i_this->mpEyeMorf->getModel();
+
+    cXyz local_24 = dComIfGp_getCamera(0)->mLookat.mEye - i_this->current.pos;
+
+    int iVar3 = cM_atan2s(local_24.x, local_24.z);
+    int iVar4 = (s16)-cM_atan2s(local_24.y, std::sqrtf(local_24.x * local_24.x + local_24.z * local_24.z));
+    
+    f32 fVar1 = 0.0f;
+    if (i_this->m2D0) {
+        fVar1 = i_this->m304;
+    }
+
+    mDoMtx_trans(mDoMtx_stack_c::get(), i_this->current.pos.x, i_this->current.pos.y + fVar1, i_this->current.pos.z);
+    
+    mDoMtx_YrotM(mDoMtx_stack_c::get(), iVar3);
+    mDoMtx_XrotM(mDoMtx_stack_c::get(), iVar4);
+    mDoMtx_ZrotM(mDoMtx_stack_c::get(), i_this->shape_angle.z + i_this->m2FA);
+
+    cXyz local_18;
+    local_18.setall(1.0f);
+
+    i_this->m2FA += i_this->m2FE;
+
+    mDoMtx_stack_c::scaleM(1.0f,1.0f,1.0f);
+
+    mDoMtx_ZrotM(mDoMtx_stack_c::get(), i_this->shape_angle.z - i_this->m2FA);
+
+    pBodyModel->setBaseTRMtx(mDoMtx_stack_c::get());
+
+    mDoMtx_trans(mDoMtx_stack_c::get(), i_this->current.pos.x, i_this->current.pos.y + fVar1, i_this->current.pos.z);
+
+    mDoMtx_YrotM(mDoMtx_stack_c::get(), i_this->shape_angle.y);
+    mDoMtx_XrotM(mDoMtx_stack_c::get(), i_this->shape_angle.x);
+    mDoMtx_ZrotM(mDoMtx_stack_c::get(), i_this->shape_angle.z);
+
+    mDoMtx_stack_c::scaleM(local_18);
+
+    pEyeModel->setBaseTRMtx(mDoMtx_stack_c::get());
+
+    enemy_fire(&i_this->mEnemyFire);
 }
 
 /* 000002CC-00000568       .text daKS_Draw__FP8ks_class */
-static BOOL daKS_Draw(ks_class*) {
-    /* Nonmatching */
+static BOOL daKS_Draw(ks_class* i_this) {
+    cXyz local_24 = i_this->current.pos - dComIfGp_getCamera(0)->mLookat.mEye;
+
+    if (local_24.abs() < REG0_F(10) + 100.0f) {
+        return TRUE;
+    }
+
+    J3DModel* pBodyModel = i_this->mpBodyMorf->getModel();
+    J3DModel* pEyeModel = i_this->mpEyeMorf->getModel();
+
+    cXyz local_18 = i_this->current.pos;
+    local_18.y += 40.0f;
+
+    dSnap_RegistFig(0xb3, i_this, 1.0f, 1.0f, 1.0f);
```

Comment:
```suggestion
    dSnap_RegistFig(DSNAP_TYPE_UNKB3, i_this, 1.0f, 1.0f, 1.0f);
```
(You can also rename `DSNAP_TYPE_UNKB3` to `DSNAP_TYPE_KS` since it's no longer unknown)